src/Apps/CollectorProfile/Routes/Saves2/Components/CreateNewListModal.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from "react"
+import { Formik } from "formik"
+import { createCollection } from "Apps/CollectorProfile/Routes/Saves2/Mutations/createCollection"
+import createLogger from "Utils/logger"
+import {
+  Flex,
+  Button,
+  LabeledInput,
+  ModalDialog,
+  Spacer,
+  EditIcon,
+  Text,
+} from "@artsy/palette"
+import { useSystemContext } from "System"
+import {
+  CreateNewListValues,
+  CreateCollectionMutationResult,
+} from "Apps/CollectorProfile/Routes/Saves2/types"
+import { Media } from "Utils/Responsive"
+import { useTranslation } from "react-i18next"
+
+interface CreateNewListModalProps {
+  visible: boolean
+  onClose: () => void
+  onComplete: (result: CreateCollectionMutationResult) => void
+}
+
+const logger = createLogger(
+  "CollectorProfile/Routes/Saves2/Components/CreateNewListModal"
+)
+
+const CreateNewListModal: React.FC<CreateNewListModalProps> = ({
+  onClose,
+  onComplete,
+}) => {
+  const { t } = useTranslation()
+  const maxNameLength = 40
+  const { relayEnvironment } = useSystemContext()
+  const [formFieldError, setFormFieldError] = useState("")
+  const [remainingCharacters, setRemainingCharacters] = useState(maxNameLength)
+
+  const handleSubmit = async (values: CreateNewListValues) => {
+    if (!relayEnvironment) {
+      return null
+    }
+
+    const params: CreateNewListValues = {
+      name: values.name,
+    }
+
+    try {
+      const response = await createCollection(relayEnvironment, params)
+      const responseOrError = response.createCollection?.responseOrError
+
+      if (responseOrError?.collection) {
+        onComplete({ id: responseOrError?.collection?.internalID })
+      } else {
+        const error = (
+          responseOrError?.mutationError?.message ?? ""
+        ).toLowerCase()
+        setFormFieldError(
+          error ?? t("collectorSaves.createNewListModal.genericError")
+        )
+      }
+    } catch (error) {
+      logger.error(error)
+      setFormFieldError(t("collectorSaves.createNewListModal.genericError"))
+    }
+  }
+
+  return (
+    <Formik<CreateNewListValues>
+      initialValues={{ name: "" }}
+      onSubmit={handleSubmit}
+    >
+      {({ values, isSubmitting, handleSubmit, handleChange, handleBlur }) => {
+        const isCreateButtonDisabled = !values.name
+
+        return (
+          <ModalDialog
+            width={["100%", 713]}
+            onClose={onClose}
+            title={t("collectorSaves.createNewListModal.title")}
+            data-testid="CreateNewList"
+            footer={
+              <>
+                {/* Desktop view */}
```

Comment:
Maybe there is a better way to handle differences in Desktop/mWeb designs, let me know

Reply:
Hard to read here, what's actually different that can't be handled simply using responsive props?

Reply:
Extra button in mobile version + button position. Dima and I tried to refactor it using responsive props but agreed that Media queries look better